### ISSUE TITLE
Consider renaming modules after #9

### ISSUE THREAD

**Issue by NHDaly** [NHDaly/ApplicationBuilder.jl#11]
PR #9 splits the Package into two main Modules, and (probably) a submodule. Here are some naming questions:

- [ ] what should we call the top-level module / package?
    - Does `ApplicationBuilder` still make sense? Should it just be `Application`? `Applications`?
- [ ] what should we call the submodule for application utilities?
    - `ApplicationBuilder.App`? `Application.Util`?  `Applications.Utils`? `ApplicationBuilder.AppUtils`?
- [ ] what should we call the second module, which contains the build code?
    - `BuildApp`? `AppBuilder`?

Here are some possible examples of how those could look:
```
# build.jl
using ApplicationBuilder; using BuildApp
build_app_bundle("hello.jl"; appname="Hello World")

# hello.jl
using ApplicationBuilder;
function Base.@ccallable julia_main(ARGS)
 ApplicationBuilder.Utils.chdir_if_bundle()
 ...
end
```

```
# build.jl
using Application; using AppBuilder;
build_app_bundle("hello.jl"; appname="Hello World")

# hello.jl
using Application;
function Base.@ccallable julia_main(ARGS)
 Application.AppUtils.chdir_if_bundle()
 ...
end
```

**Comment by NHDaly** [comment]
From https://github.com/NHDaly/ApplicationBuilder.jl/pull/9#issuecomment-388644309:

> No, I think the package can stay as `ApplicationBuilder`. I think the submodule with `build_app_bundle` should be `Builder`. And `change_dir_if_bundle` can stay in `BuildUtils`.

**Comment by NHDaly** [comment]
> `build_app_bundle` should be in `Builder`
@ranjanan So the problem with having two modules like this, is that the second module _isn't_ a submodule, so its name has to stay unique across all julia modules. Imagine it basically as a totally separate package _squished_ into this package. So I think `Builder.jl` would almost certainly get push-back as too generic.

Regarding `BuildUtils`, I very specifically want to keep "build" out of the name, since it contains _runtime_ utilities, not build-time utils. Do you prefer `ApplicationBuilder.AppUtils` over `ApplicationBuilder.Utils` or `ApplicationBuilder.App`?

**Comment by NHDaly** [comment]
(For now, i've stuck with `ApplicationBuilder.App`, in order to hint towards the fact that this is the code that an actual _application_ should use.)

**Comment by ranjanan** [comment]
I think `ApplicationBuilder.App` is fine too. Let's call the module where `build_app_bundle` resides as `App`, and perhaps where `change_dir_if_bundle` resides as `ApplicationBuilder.RuntimeUtils`?

**Comment by rapus95** [comment]
came here from the JuliaCon talk, was looking for "autosnoop" documentation, and for this issue, what about "Helper" instead of "Util"? in case it's not already to late anyway :D

**Comment by NHDaly** [comment]
Haha thanks! Actually, thankfully, I'm pretty sure I can finally just delete all these modules!
They'll be deleted in https://github.com/NHDaly/ApplicationBuilder.jl/pull/18. :)

Thanks for the suggestion though, I do agree that "Helper" would probably have been a better name.

And oh, sorry, yeah about the lack of documentation on `autosnoop`. It was a very last-minute thing... I'm still not sure if it's the right solution, or if we should just be changing the compiler settings like Jeff suggested during the talk. I'll need to look into it more!

**Comment by NHDaly** [comment]
I've created a new Issue here to discuss "complete compilation" further: https://github.com/NHDaly/ApplicationBuilder.jl/issues/24

(That said, for now, the api for `autosnoop` is just to pass it to `build_app_bundle` like this: `build_app_bundle(f, autosnoop=true)`, but i actually don't _really_ know if it's making a difference. Haha sorry!
